pkg/kubectl/pdb.go

```diff
@@ -66,6 +67,11 @@ func (s PodDisruptionBudgetV1Generator) Generate(params map[string]interface{})
 
 // StructuredGenerate outputs a pod disruption budget object using the configured fields.
 func (s *PodDisruptionBudgetV1Generator) StructuredGenerate() (runtime.Object, error) {
+	if len(s.MinAvailable) == 0 {
```

Comment:
The behavior is slightly different from before.
Before: user specifies `MinAvailable=0`, we error out.
After: user specifies `MinAvailable=0`, we change it to 1.

A way to fix this is
- add function `Complete()` in `create_pdb.go`, [ref](https://github.com/kubernetes/community/blob/master/contributors/devel/kubectl-conventions.md#command-implementation-conventions). And do the defaulting there.
- There is a pflag function [Changed](https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/spf13/pflag/flag.go#L401) that will be helpful for you to determine if the user has specified a flag. With it you can distinguish unset or 0.

With these change you don't need to defaulting in both of these 2 generators.